ups_modul/inout/inout_gsheet.py

```diff
@@ -3,178 +3,235 @@
-def SUM_hours(days_range, wks):
-    for keys in anthill_name:
-        a1_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row,2)
-        a2_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+1)
-        wks.update_cell(anthill_name[keys].row, days_range+2, "=SUM({}:{})".format(a1_notion, a2_notion))
-
-# SUM hours and pay
-def SUM_wage(days_range, wks):
-    for keys in anthill_name:
-        a1_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+2)
-        wks.update_cell(anthill_name[keys].row, days_range+3 ,"=SUM({}*{})".format(a1_notion, anthill_name[keys].pay))
-
-# SUM 99% of the wage
-def SUM_wage99(days_range, wks):
-    for keys in anthill_name:
-        a1_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+3)
-        wks.update_cell(anthill_name[keys].row, days_range+4 ,"=0.99*ROUNDDOWN({};0)".format(a1_notion))
-
-# SUM tips ratio for each row user time to all time
-def SUM_tips_ratio(days_range, wks):
-    a2_notion = gspread.utils.rowcol_to_a1(2+anthill.num_of_ants, days_range+2)
-    for keys in anthill_name:
-        a1_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+2)
-        wks.update_cell(anthill_name[keys].row, days_range+5 ,"={}/{}".format(a1_notion, a2_notion))
-
-# SUM tips based on ratio
-def SUM_tips(days_range, wks):
-    a2_notion = gspread.utils.rowcol_to_a1(2+anthill.num_of_ants, days_range+6)
-    for keys in anthill_name:
-        a1_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+5)
-        wks.update_cell(anthill_name[keys].row, days_range+6 ,"=ROUNDDOWN({}*{};0)".format(a1_notion, a2_notion))
-
-# SUM total of the payment
-def SUM_total(days_range, wks):
-    for keys in anthill_name:
-        a1_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+4)
-        a2_notion = gspread.utils.rowcol_to_a1(anthill_name[keys].row, days_range+6)
-        wks.update_cell(anthill_name[keys].row, days_range+7 ,"=ROUNDDOWN({}+{};0)".format(a1_notion, a2_notion))
-
-# writing the user time in the user cell in the day IN
-def user_time_log(day_in, user_name, hours, wks):
+def names_rows(anthill, wks):
+    '''
+    names from the anthill list in rows
+    '''
+    names_list = [] 
+    a1_notion, a2_notion = a_notion(2, 1, anthill.num_of_ants+1, 1)
+    cell_list = wks.range("{}:{}".format(a1_notion, a2_notion))
+    names_list = [key for key, values in sorted(anthill_name.items(), key=lambda x: x[1].row)]
+    batch_cells(names_list, cell_list, wks)
+
+def header_days(days, wks):
+    '''
+    writing month's days in header row
+    '''
+    days_list = []
+    a1_notion, a2_notion = a_notion(1, 2, 1, days+1)
+    cell_list = wks.range("{}:{}".format(a1_notion, a2_notion))
+    for x in range(1, days+1):
+        days_list.append(x)
+    batch_cells(days_list, cell_list, wks)
+
+def header_calculation(days, wks):
+    '''
+    header labels for calculation
```

Comment:
Function header not clear

Reply:
labels for the columns in the SUM part of the worksheets

Reply:
Try using verbs.

header_days():
'''
Writes the days of the month in the header row of the spreadsheet

'''